### PR TITLE
[Storage] `az storage share-rm`: Support create/show/delete snapshots

### PR DESCRIPTION
**Pull request by evelyn-ys**
**Description**

File share management plane operations support snapshot CRD. (_List operation is not covered in this PR because list including both deleted and snapshots is not supported which is a known issue will be resolved in next api version._)

Related swagger: https://github.com/Azure/azure-rest-api-specs/pull/11372

**Testing Guide**

Create: `az storage share-rm snapshot`
Show: `az storage share-rm show --snapshot`
Delete: `az storage share-rm delete --snapshot`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Storage

File: src/azure-cli/azure/cli/command_modules/storage/_help.py [hunk 1]
```diff
@@ -2253,14 +2253,16 @@
...
-short-summary: Delete the specified Azure file share.
+short-summary: Delete the specified Azure file share or share snapshot.
 examples:
   - name: Delete an Azure file share 'myfileshare' under the storage account 'mystorageaccount' (account name) in resource group 'MyResourceGroup'.
     text: az storage share-rm delete -g MyResourceGroup --storage-account mystorageaccount --name myfileshare
   - name: Delete an Azure file share 'myfileshare' under the storage account 'mystorageaccount' (account id).
     text: az storage share-rm delete --storage-account mystorageaccount --name myfileshare
   - name: Delete an Azure file share by resource id.
     text: az storage share-rm delete --ids file-share-id
+  - name: Delete an Azure file share snapshot.
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/storage/_help.py:2264]: We are hesitated to add delete command because it may have breaking change.

Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/_help.py:2264]: Discussed with yishi offline, this is for deletion of specific snapshot.  Breaking change is for delete share with all snapshots.

File: src/azure-cli/azure/cli/command_modules/storage/_help.py [hunk 2]
```diff
@@ -2329,6 +2333,14 @@
     text: az storage share-rm update --ids file-share-id --quota 3 --metadata key1=value1 key2=value2
 """
 
+helps['storage share-rm snapshot'] = """
+type: command
+short-summary: Creates a snapshot of an existing share under the specified account.
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/storage/_help.py:2338]:
```suggestion
short-summary: Create a snapshot of an existing share under the specified account.
```

File: src/azure-cli/azure/cli/command_modules/storage/commands.py [hunk 3]
```diff
@@ -501,15 +501,18 @@ def get_custom_sdk(custom_module, client_factory, resource_type=ResourceType.DAT
...
         g.command('delete', 'delete', confirmation=True)
         g.custom_command('exists', '_file_share_exists', transform=create_boolean_result_output_transformer('exists'))
         g.custom_command('list', 'list_share_rm')
-        g.show_command('show', 'get')
+        g.show_command('show', 'get', transform=transform_share_rm_output)
         g.generic_update_command('update', setter_name='update', setter_arg_name='file_share',
                                  custom_func_name='update_share_rm')
         g.custom_command('stats', 'get_stats', transform=lambda x: getattr(x, 'share_usage_bytes'))
         g.custom_command('restore', 'restore_share_rm')
+        g.custom_command('snapshot', 'snapshot_share_rm', min_api='2020-08-01-preview',
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/storage/commands.py:514]: could you add preview tag for new commands?

File: src/azure-cli/azure/cli/command_modules/storage/_help.py (hunk 2 above)
Comment (evelyn-ys) [reply src/azure-cli/azure/cli/command_modules/storage/_help.py]: Done. Thanks!

File: src/azure-cli/azure/cli/command_modules/storage/commands.py (hunk 3 above)
Comment (evelyn-ys) [reply src/azure-cli/azure/cli/command_modules/storage/commands.py]: Added for new command and new parameter.

**Review by Juliehzl — APPROVED** [review]

**Pull request merged**